# Netflix Financial Analysis Project

This Jupyter Notebook aims to analyze Netflix's financial data to gain insights into the company's revenue trends and regional revenue distribution over the years. The project seeks to address the following research questions:

1. What are the quarterly and yearly revenue trends for Netflix from 2012 to 2023?
2. How does revenue contribution vary across different regions (UACN, EMEA, LATM, APAC) from 2019 to 2023?
3. What is the proportion of revenue generated by each region during the specified period?

The analysis will involve visualization, and interpretation of key findings to support decision-making and strategic planning for Netflix's future growth and expansion strategies.


In [16]:
# Total revenue per year and per quarter

# Import necessary libraries
import pandas as pd
import plotly.express as px

# Load the DataFrame from the CSV file
path = r'C:\Users\prett\OneDrive\Desktop\CODING\PYTHON\ANALYSIS\kaggleNetflixRevenue.csv'
df = pd.read_csv(path)

# Convert the 'Date' column to datetime type with the specified format
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)

# Extract the year and quarter from the 'Date' column
df['Year'] = df['Date'].dt.year
df['Quarter'] = df['Date'].dt.quarter

# Calculate the total revenue for each quarter using groupby and transform
df['Total Revenue'] = df.groupby(['Year', 'Quarter'])['Global Revenue'].transform('sum')

# Print the DataFrame for quarterly revenue
quarterly_revenue_df = df.groupby(['Year', 'Quarter'])['Global Revenue'].sum().reset_index()
print("Quarterly Revenue DataFrame:")
print(quarterly_revenue_df)

# Print the DataFrame for yearly revenue
yearly_revenue_df = df.groupby('Year')['Total Revenue'].sum().reset_index()
print("\nYearly Revenue DataFrame:")
print(yearly_revenue_df)


Quarterly Revenue DataFrame:
    Year  Quarter  Global Revenue
0   2012        1       869791000
1   2012        2       889163000
2   2012        3       905089000
3   2012        4       945239000
4   2013        1      1023961000
5   2013        2      1069372000
6   2013        3      1105999000
7   2013        4      1175230000
8   2014        1      1270089000
9   2014        2      1340407000
10  2014        3      1409432000
11  2014        4      1484728000
12  2015        1      1573129000
13  2015        2      1644694000
14  2015        3      1738355000
15  2015        4      1823333000
16  2016        1      1957736000
17  2016        2      2105204000
18  2016        3      2290188000
19  2016        4      2477541000
20  2017        1      2636635000
21  2017        2      2785464000
22  2017        3      2984859000
23  2017        4      3285755000
24  2018        1      3700856000
25  2018        2      3907270000
26  2018        3      3999374000
27  2018        4  

In [11]:
# Plotting the yearly revenue trend with a line chart
fig_yearly_revenue = px.line(yearly_revenue_df, x='Year', y='Total Revenue', title='Netflix Yearly Revenue from 2012 to 2023')
fig_yearly_revenue.show()

# Plotting the quarterly revenue trend with a bar chart
fig_quarterly_revenue = px.bar(quarterly_revenue_df, x='Year', y='Global Revenue', color='Quarter', title='Netflix Quarterly Revenue from 2012 to 2023')
fig_quarterly_revenue.show()

Netflix's Quarterly and Yearly Revenue trends show consistent growth, with each quarter exceeding the previous year's revenue. The company follows a seasonal pattern, increasing from the first to the fourth quarter each year. The highest quarterly revenue is typically recorded in the fourth quarter, aligning with the holiday season.

The Yearly Revenue trend shows steady upward growth from 2012 to 2023, indicating sustained and substantial growth in Netflix's business. The growth rate is not uniform but shows an upward trend with some fluctuations in certain years. Netflix experienced a significant jump in Yearly Revenue from 2016 to 2017, indicating rapid expansion.

While the Yearly Revenue growth in 2023 is lower than previous years, it is important to note that this data only represents the first quarter of 2023. Additional data for the remaining quarters of 2023 would be needed to assess the complete trend. Overall, Netflix's consistent growth in both Quarterly and Yearly Revenues demonstrates its strong position in the streaming industry and ability to attract and retain subscribers.

In [12]:
#Revenue for each region from 2019 to 2023 and a pie chart showing this

# Select the relevant columns for revenue and year
regional_revenue_df = df[['Year', 'UACN Revenue', 'EMEA Revenue', 'LATM Revenue', 'APAC Revenue']]

# Filter data from 2019 to 2023
regional_revenue_df = regional_revenue_df[regional_revenue_df['Year'].between(2019, 2023)]


# Group by year and sum the revenue columns for each region
regional_revenue_per_year = regional_revenue_df.groupby('Year').sum()


print("Revenue for Each Region from 2019 to 2023:")
print(regional_revenue_per_year)



Revenue for Each Region from 2019 to 2023:
      UACN Revenue  EMEA Revenue  LATM Revenue  APAC Revenue
Year                                                        
2019  1.005121e+10  5.543067e+09  2.795434e+09  1.469521e+09
2020  1.145540e+10  7.772252e+09  3.156727e+09  2.372300e+09
2021  1.297210e+10  9.699819e+09  3.576976e+09  3.266601e+09
2022  1.408464e+10  9.745015e+09  4.069973e+09  3.570221e+09
2023  3.608645e+09  2.517641e+09  1.070192e+09  9.335230e+08


In the year 2023, the revenue for the EMEA region is approximately 2.52 billion USD, while the revenue for the UACN region is around 3.61 billion USD. Therefore, in 2023, the UACN region's revenue is still higher than the revenue of the EMEA region.

However, there is a notable trend that occurred between 2021 and 2022. In 2021, the revenue contribution of the EMEA region was approximately 9.70 billion USD, and in 2022, it increased to around 9.75 billion USD. During this time, the UACN region's revenue contribution was approximately 13.00 billion USD in 2021 and increased to around 14.08 billion USD in 2022.

In 2022, the revenue contribution of the EMEA region came very close to that of the UACN region, and there was a smaller gap in their respective revenue contributions. While the EMEA region did not surpass the UACN region in 2022, it demonstrated substantial growth and narrowed the revenue difference between the two regions during that period.

In [13]:
columns_to_sum = ['UACN Revenue', 'EMEA Revenue', 'LATM Revenue', 'APAC Revenue']
revenue_contribution = regional_revenue_per_year[columns_to_sum].sum()
# Create a DataFrame with the revenue contribution data
data = pd.DataFrame({
    'Region': revenue_contribution.index,
    'Revenue': revenue_contribution.values
})

# Plot the pie chart
fig = px.pie(data, values='Revenue', names='Region', title='Revenue Contribution by Region')

# Show the pie chart
fig.show()


Based on the pie chart, the proportion of revenue generated by each region during the specified period (2019 to 2023) can be summarised as follows:

EMEA Region: The EMEA region contributed approximately 31% of the total revenue generated by Netflix during the specified period.

LATM Region: The LATM region contributed approximately 12.9% of the total revenue.

APAC Region: The APAC region contributed approximately 10.2% of the total revenue.

UACN Region: The UACN region (United States and Canada) had the highest revenue contribution, accounting for approximately 45.9% of the total revenue.

These percentages provide insights into the revenue distribution among different regions and illustrate the significance of each region in contributing to Netflix's overall revenue during the specified period.

# Key Findings and Insights

The revenue trends between the EMEA and UACN regions provide valuable business insights for Netflix. The steady growth in the EMEA region from 2019 to 2022 indicates untapped market opportunities, which Netflix could capitalize on by investing in content localization, marketing, and customer engagement strategies. The region's revenue approaching that of the UACN region in 2022 demonstrates the potential for international markets to drive Netflix's growth. Diversifying revenue sources is crucial, as overreliance on a single region exposes Netflix to market-specific risks and fluctuations. Understanding the factors behind revenue trends in different regions can provide valuable insights into the drivers of success and challenges in each market.

Analyzing revenue trends can help Netflix optimize content licensing and production decisions, identifying content that resonates well in specific markets, and determining appropriate subscription pricing in each region. Strategic resource allocation is also essential, as identifying regions with strong revenue growth potential can lead to more efficient resource utilization and higher returns on investment. By leveraging these insights, Netflix can adapt and thrive in a dynamic global streaming market and maintain its position as a leading entertainment provider.